File: 3838/solution.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        Map<Character, Integer> cmap = new HashMap<>();
        Map<Integer, Character> vmap = new HashMap<>();
        char c;
        int curr;
        String res = "";
        for(int i=0;i<26;i++){
            c =(char) (97+i);
            cmap.put(c, weights[i]);
            vmap.put(25-i,c);
        }
        System.out.println(vmap);

        for(String word: words){
            curr = 0;
            for(int i=0;i<word.length();i++)
                curr+=cmap.get(word.charAt(i));
            res+=vmap.get(curr%26);
        }
        return res;  
    }
}

File: 3838/solution.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        di = {}
        ci = {}
        for i in range(26):
            di[chr(97+i)] = weights[i]
            ci[25-i] = chr(97+i)
        res = ''
        for word in words:
            curr = 0
            for c in word:
                curr+=di[c]
            res+=ci[curr%26]
        return res
